pkg/apis/authentication/v1alpha1/types_credentialsbinding.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CredentialsBinding represents a binding to credentials in the same or another namespace.
+type CredentialsBinding struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata.
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Provider defines the provider type of the CredentialsBinding.
+	// This field is immutable.
+	Provider CredentialsBindingProvider `json:"provider" protobuf:"bytes,2,opt,name=provider"`
+	// CredentialsRef is a reference to a resource holding the credentials.
+	// Accepted resources are core/v1.Secret and authentication.gardener.cloud/v1alpha1.WorkloadIdentity
+	CredentialsRef corev1.ObjectReference `json:"credentialsRef" protobuf:"bytes,3,name=credentialsRef"`
+	// Quotas is a list of references to Quota objects in the same or another namespace.
+	// This field is immutable.
+	// +optional
+	Quotas []corev1.ObjectReference `json:"quotas,omitempty" protobuf:"bytes,4,rep,name=quotas"`
+}
+
+// GetProviderType gets the type of the provider.
+func (cb *CredentialsBinding) GetProviderType() string {
```

Comment:
What is the purpose of this function?

Reply:
Implementing the interface https://github.com/gardener/gardener/blob/0e40e4a6b381ea1d9278e47794388f924f03487b/pkg/apis/core/types.go#L17-L22, I am not completely sure, but I guess it might be required for extensions predicate here https://github.com/gardener/gardener/blob/0e40e4a6b381ea1d9278e47794388f924f03487b/extensions/pkg/predicate/predicate.go#L117

Reply:
Aha, thanks!